sycl/source/detail/memory_manager.cpp

```diff
@@ -162,7 +181,7 @@ void *MemoryManager::allocateMemImage(
 void *MemoryManager::allocateMemSubBuffer(ContextImplPtr TargetContext,
                                           void *ParentMemObj, size_t ElemSize,
                                           size_t Offset, range<3> Range,
-                                          std::vector<RT::PiEvent> DepEvents,
+                                          std::vector<EventImplPtr> DepEvents,
```

Comment:
Why do you change interface?

Reply:
But it seems you already extract plugin from queue/context which can be passed to waitForEvents.